java-frontend/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java

```diff
@@ -225,4 +241,30 @@ public int hashCode() {
       return Objects.hash(msg, syntaxNode);
     }
   }
+
+  /**
+   * Cognitive complexity and associated locations
+   */
+  class CognitiveComplexity {
+    public static final CognitiveComplexity EMPTY = new CognitiveComplexity(0, Collections.emptyList());
+
+    /**
+     * Numerical value corresponding to the cognitive complexity
+     */
+    public final int complexity;
+    /**
+     * Secondary locations related to the cognitive complexity nodes
+     */
+    public final List<Location> locations;
+
+    public CognitiveComplexity(int complexity, List<Location> locations) {
+      this.complexity = complexity;
+      this.locations = locations;
+    }
+
+    public static CognitiveComplexity empty() {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') Rename method "empty" to prevent any misunderstanding/clash with field "EMPTY" defined on line 249. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1845)